File: problems/algorithms/array-prime-factors/ruby/code/prime-factors.spec.rb
require_relative 'prime-factors.rb'

describe 'Prime Factors' do
  context 'returning an empty array for 1' do
    it [] do
      expect(prime_factors(1)).toEqual([])
    end
  end

  context 'factoring 2' do
    it [2] do
      expect(prime_factors(2)).toEqual([2])
    end
  end

  context 'factoring 3' do
    it [3] do
      expect(prime_factors(3)).toEqual([3])
    end
  end

  context 'factoring 4' do
    it [2, 2] do
      expect(prime_factors(4)).toEqual([2, 2])
    end
  end

  context 'factoring 6' do
    it [2, 3] do
      expect(prime_factors(6)).toEqual([2, 3])
    end
  end

  context 'factoring 8' do
    it [2, 2, 2] do
      expect(prime_factors(8)).toEqual([2, 2, 2])
    end
  end

  context 'factoring 9' do
    it [3, 3] do
      expect(prime_factors(9)).toEqual([3, 3])
    end
  end

  context 'factoring 27' do
    it [3, 3, 3] do
      expect(prime_factors(27)).toEqual([3, 3, 3])
    end
  end

  context 'factoring 625' do
    it [5, 5, 5, 5] do
      expect(prime_factors(625)).toEqual([5, 5, 5, 5])
    end
  end

  context 'factoring 901255' do
    it [5, 17, 23, 461] do
      expect(prime_factors(901255)).toEqual([5, 17, 23, 461])
    end
  end

  context 'factoring 93819012551' do
    it [11, 9539, 894119] do
      expect(prime_factors(93819012551)).toEqual([11, 9539, 894119])
    end
  end
end

File: src/pages/exercises/array-prime-factors/_codes/php/response/spec/prime-factors.spec.php
<?php
require __DIR__ . '/../src/prime-factors.php';

describe('Prime Factors', function () {
  it('returning an empty array for 1', function () {
    expect(primeFactors(1))->toEqual([]);
  });

  it('factoring 2', function () {
    expect(primeFactors(2))->toEqual([2]);
  });

  it('factoring 3', function () {
    expect(primeFactors(3))->toEqual([3]);
  });

  it('factoring 4', function () {
    expect(primeFactors(4))->toEqual([2, 2]);
  });

  it('factoring 6', function () {
    expect(primeFactors(6))->toEqual([2, 3]);
  });

  it('factoring 8', function () {
    expect(primeFactors(8))->toEqual([2, 2, 2]);
  });

  it('factoring 9', function () {
    expect(primeFactors(9))->toEqual([3, 3]);
  });

  it('factoring 27', function () {
    expect(primeFactors(27))->toEqual([3, 3, 3]);
  });

  it('factoring 625', function () {
    expect(primeFactors(625))->toEqual([5, 5, 5, 5]);
  });

  it('factoring 901255', function () {
    expect(primeFactors(901255))->toEqual([5, 17, 23, 461]);
  });

  it('factoring 93819012551', function () {
    expect(primeFactors(93819012551))->toEqual([11, 9539, 894119]);
  });
});
